lib/luarcu.c

```diff
@@ -177,6 +177,19 @@ lunatik_object_t *luarcu_gettable(lunatik_object_t *table, const char *key, size
 }
 EXPORT_SYMBOL(luarcu_gettable);
 
+int luarcu_settable(lunatik_object_t *table, const char *key, size_t keylen, lunatik_object_t *object)
+{
+	int ret;
+
+	lunatik_lock(table);
+	ret = luarcu_insert(table->private, key, keylen, object);
+	lunatik_unlock(table);
+	if (!rcu_read_lock_held())
+		synchronize_rcu();
+	return ret;
```

Comment:
I think returning `-ENOMEM` would be more meaningful for the API client.. perhaps the best place to add it is here:

https://github.com/luainkernel/lunatik/pull/118/files#diff-87fd494e0e522758b53f3d5ec4a3c8e9886f1631f5ed1efb61a56755a22ba37dR115

and here:

https://github.com/luainkernel/lunatik/pull/118/files#diff-87fd494e0e522758b53f3d5ec4a3c8e9886f1631f5ed1efb61a56755a22ba37dR120